src/helm/benchmark/model_deployment_registry.py

```diff
@@ -104,5 +134,152 @@ def maybe_register_model_deployments_from_base_path(base_path: str) -> None:
         register_model_deployments_from_path(path)
 
 
-def get_model_deployment(name: str) -> Optional[ModelDeployment]:
-    return _name_to_model_deployment.get(name)
+# ===================== UTIL FUNCTIONS ==================== #
+def get_model_deployment(name: str) -> ModelDeployment:
+    register_deployments_if_not_already_registered()
+    if name not in DEPLOYMENT_NAME_TO_MODEL_DEPLOYMENT:
+        raise ValueError(f"Model deployment {name} not found")
+    deployment: ModelDeployment = DEPLOYMENT_NAME_TO_MODEL_DEPLOYMENT[name]
+    if deployment.deprecated:
+        hlog(f"WARNING: Model deployment {name} is deprecated")
+    return deployment
+
+
+# TODO: Figure out when we can remove this.
+#
+# This is used in 2 contexts:
+# 1. Backward compatibility. We can remove this when we no longer want to offer backwards compatibility
+#    for model names that are now moved to model deployments (PR #1903).
+# 2. Support use cases like "model=text". It was decided that "model_deployment=text" would not be
+#    supportedfor now as it coudl evaluate several deployments for the same model. This use case will
+#    still need to be supported after we remove backward compatibility.
+
+
+def get_deployment_name_from_model_arg(
+    model_arg: str, can_return_empy: bool = False, warn_arg_deprecated: bool = True
+) -> str:
+    """Returns a valid model deployment name corresponding to the given model arg.
+    This is used as a backwards compatibility layer for model names that are now moved to model deployments.
+    Example: "anthropic/claude-v1.3" => "anthropic/claude-v1.3"
+    Example: "meta/llama-7b" => "together/llama-7b"
+
+    The process to find a model deployment name is as follows:
+    1. If there is a model deployment with the same name as the model arg, use it.
+    2. If there is at least one deployment for the model, use the first one that is available.
+    3. If there are no deployments for the model, raise an error.
+
+    This function will also try to find a model deployment name that is not deprecated.
+    If there are no non-deprecated deployments, it will return the first deployment (even if it's deprecated),
+    unless can_return_empy is True, in which case it will return an empty string.
+
+    If warn_arg_deprecated is True, this function will print a warning if the model deployment name is not the same
+    as the model arg. This is to remind the user that the model name is deprecated and should be replaced with
+    the model deployment name (in their config).
+
+    Args:
+        model_arg: The model arg to convert to a model deployment name.
+        can_return_empy: Whether to return an empty string if some model deployments were found but all are deprecated.
+        warn_arg_deprecated: Whether to print a warning if the model deployment name is not the same as the model arg.
+    """
+
+    # Register model deployments if not already registered.
+    register_deployments_if_not_already_registered()
```

Comment:
Why is this everywhere? Just call this once in the `run.py` main function.

Reply:
Some calls do not go through `run.py` or `summarize.py` and so this hasn't been called yet (test for example and some weird edge cases that I was not able to figure out). This should be cheap enough I think (just a condition) and ensures that everything is registered properly